github-adapter/src/main/java/be/xplore/githubmetrics/githubadapter/config/ratelimiting/RateLimitResetAwaitScheduler.java

```diff
@@ -24,20 +24,22 @@ public RateLimitResetAwaitScheduler(
     }
 
     public void createStopAllRequestsTask(
-            Runnable reset, Duration restartTime
+            Runnable reset, Instant restartTime
     ) {
-        this.stopEvicting = Optional.of(this.taskScheduler.scheduleWithFixedDelay(
-                () -> {
-                    reset.run();
-                    this.stopEvicting.get().cancel(false);
-                    this.stopEvicting = Optional.empty();
-                    LOGGER.trace("Finished reset await task. Remove the future.");
-                },
-                restartTime
-        ));
-        LOGGER.debug(
-                "Reset await task has been created, will run reset in {}s.",
-                restartTime.toSeconds()
-        );
+        if (this.stopEvicting.isEmpty()) {
+            this.stopEvicting = Optional.of(this.taskScheduler.schedule(
+                    () -> {
+                        reset.run();
+                        this.stopEvicting.get().cancel(false);
+                        this.stopEvicting = Optional.empty();
+                        LOGGER.info("Finished reset await task. Remove the future.");
+                    },
+                    restartTime
+            ));
+            LOGGER.warn(
+                    "Reset await task has been created, will run reset at {}.",
+                    restartTime
+            );
```

Comment:
Maybe say something that is a little clearer. `Reset await task` seems a bit non descriptive in a log context